R/doeFactorial.R

```diff
@@ -175,100 +181,200 @@ doeFactorial <- function(jaspResults, dataset, options, ...) {
+                                repeat.only = rep,
                                 randomize = rnd)
-        runOrder <- 1:nrow(splitRuns)
-        rows <- data.frame(runOrder = runOrder)
-        rows <- cbind.data.frame(rows,
-                                 splitRuns)
       }
-    } #blocks NO
+      runOrder <- 1:nrow(splitRuns)
+      rows <- cbind.data.frame(runOrder, splitRuns)
+
+      table$addFootnote(paste("Hard-to-change factors: ", paste(factorNames[1:options[["numberHTCFactors"]]], collapse = ", "), sep = ""))
+    }
 
+    #full
     if (options[["factorialType"]] == "factorialTypeFull") {
-      runOrder <- 1:2^options[["numberOfFactors"]]
-      rows <- data.frame(runOrder = runOrder)
-      rows <- cbind.data.frame(rows,
-                               FrF2::FrF2(nfactors = options[["numberOfFactors"]],
-                                          nruns = 2^options[["numberOfFactors"]],
-                                          randomize = rnd))
-    } #blocks YES
+      rows <- FrF2::FrF2(nfactors = options[["numberOfFactors"]],
+                         nruns = 2^options[["numberOfFactors"]],
+                         ncenter = options[["factorialCenterPoints"]],
+                         replications = options[["factorialCornerReplicates"]],
+                         repeat.only = rep,
+                         blocks = options[["factorialBlocks"]],
+                         randomize = rnd)
+      runOrder <- 1:nrow(rows)
+      rows <- cbind.data.frame(runOrder, rows)
+    }
+
+    #p-b
+    if(options[["factorialType"]] == "factorialPlackettBurman"){
+      rows <- FrF2::pb(nruns = as.numeric(options[["PBruns"]]),
+                       nfactors = options[["numberOfFactors"]],
+                       ncenter = options[["factorialCenterPoints"]],
+                       replications = options[["factorialCornerReplicates"]],
+                       repeat.only = rep,
+                       randomize = rnd)
+      runOrder <- 1:nrow(rows)
+      rows <- cbind.data.frame(runOrder, rows)
+    }
 
+    blocks <- rows$Blocks
+    rows <- rows[,!names(rows) %in% "Blocks"]
+
+    #filling in table
     for (i in 1:as.numeric(options[["numberOfFactors"]])) {
       colnames(rows)[i+1] <- factorNames[i]
       table$addColumnInfo(name = factorNames[i], title = factorNames[i], type = 'string')
     }
 
-    jaspResults[["displayDesign"]] <- table
+    if(options[["factorialBlocks"]] > 1 || options[["factorialCornerReplicates"]] > 1){
+      blocks <- rows$Blocks
+      rows <- rows[,!names(rows) %in% "Blocks"]
+      # table$addColumnInfo(name = 'blocks', title = gettext("Blocks"), type = 'string')
+
+      if(options[["factorialBlocks"]] == 1 && options[["factorialCornerReplicates"]] > 1 && options[["factorialCenterPoints"]] > 0){
+        for(i in 1:options[["factorialCornerReplicates"]]){
+          for(j in 1:options[["factorialCenterPoints"]]){
```

Comment:
So in QML you changed the minimum value people can enter to `0` as a value for `factorialCenterPoints`. Then you would get `1:0` here which might not be what you want. In general, if 0 leads to errors it's always safer to do:
```suggestion
        for(i in seq_len(options[["factorialCornerReplicates"]])){
          for(j in seq_len(options[["factorialCenterPoints"]])){
```
as `seq_len(0)` returns `integer(0)` and the body of the loop won't be run.

Reply:
✅